File: Dataset/Leetcode/train/13/477.java
 class Solution {
    public int XXX(String s) {
        Map<Character, Integer> map = new HashMap<>();
        map.put(' ', 0);
        map.put('I', 1);
        map.put('V', 5);
        map.put('X', 10);
        map.put('L', 50);
        map.put('C', 100);
        map.put('D', 500);
        map.put('M', 1000);
        
        char prevCh = ' ';
        
        int ans = 0;
        
        for (int i = s.length() - 1; i >= 0; i--) {
            char currCh = s.charAt(i);
            int curr = map.get(currCh);
            if (map.get(prevCh) > curr) {
                ans -= curr;
            } else {
                ans += curr;
            }
            prevCh = currCh;
        }
        
        return ans;
    }
}

File: Dataset/Leetcode/test/62/477.cpp
class Solution {
public:
    int f[110][110];
    
    int XXX(int m, int n) {
        
        for(int i=0;i<m;i++)
            for(int j=0;j<n;j++)
                if(i == 0 && j == 0) f[i][j] = 1;
                else if(i == 0 && j !=0) f[i][j] = f[i][j-1] ;
                else if(i!=0 && j ==0) f[i][j] = f[i-1][j];
                else f[i][j] = f[i-1][j] + f[i][j-1];
        return f[m-1][n-1];
    }
};

